cmd/cloudFoundryDeploy.go

```diff
@@ -226,46 +218,30 @@ func handleMTADeployment(config *cloudFoundryDeployOptions, command command.Exec
 }
 
 type deployConfig struct {
-	DeployCommand   string
-	DeployOptions   []string
-	AppName         string
-	ManifestFile    string
-	SmokeTestScript []string
+	DeployCommand string
+	DeployOptions []string
+	AppName       string
+	ManifestFile  string
 }
 
 func handleCFNativeDeployment(config *cloudFoundryDeployOptions, command command.ExecRunner) error {
 
-	deployType, err := checkAndUpdateDeployTypeForNotSupportedManifest(config)
-
-	if err != nil {
-		return err
-	}
-
 	var deployCommand string
-	var smokeTestScript []string
 	var deployOptions []string
 
 	// deploy command will be provided by the prepare functions below
-
-	if deployType == "blue-green" {
-		log.Entry().Warn("[WARN] Blue-green deployment type is deprecated for cf native builds " +
-			"and will be completely removed by 15.06.2024" +
+	if config.DeployType == "blue-green" {
+		return fmt.Errorf("Blue-green deployment type is deprecated for cf native builds." +
 			"Instead set parameter `cfNativeDeployParameters: '--strategy rolling'`. " +
 			"Please refer to the Cloud Foundry documentation for further information: " +
 			"https://docs.cloudfoundry.org/devguide/deploy-apps/rolling-deploy.html." +
 			"Or alternatively, switch to mta build tool. Please refer to mta build tool" +
 			"documentation for further information: https://sap.github.io/cloud-mta-build-tool/configuration/.")
-		deployCommand, deployOptions, smokeTestScript, err = prepareBlueGreenCfNativeDeploy(config)
-		if err != nil {
-			return errors.Wrapf(err, "Cannot prepare cf native deployment. DeployType '%s'", deployType)
-		}
-	} else if deployType == "standard" {
-		deployCommand, deployOptions, smokeTestScript, err = prepareCfPushCfNativeDeploy(config)
-		if err != nil {
-			return errors.Wrapf(err, "Cannot prepare cf push native deployment. DeployType '%s'", deployType)
-		}
-	} else {
-		return fmt.Errorf("Invalid deploy type received: '%s'. Supported values: %v", deployType, []string{"blue-green", "standard"})
+	}
```

Comment:
Let's keep the old if else structure, as there might be somebody with a typo, or using the "bg-deploy" deployType [which works for mta (although it is not documented)](https://github.com/SAP/jenkins-library/blob/master/cmd/cloudFoundryDeploy.go#L675-L677) here and expects a zero-downtime deployment.

Reply:
I do not see `bg-deploy` used for old if-else of cf_native deployment. Moreover, that command is used w.r.t mta Multiapps plugin. Do you think it still makes sense to add it for cf_native?

Reply:
Ok, let me formulate it differntly. What we had before was a check that somebody chose "standard"; now we let everything through, except "blue-green". I would like to block anything but "standard" now. For "blue-green" we have this special message. For anything else we tell the user we don't know what they want, only "standard" is available as an option.

Reply:
I want the user to be aware of choosing standard, rather than writing "blue-gren" and expecting zero downtime. We blocked such typos in the past, we should still block them in the future. Only "standard" is allowed here.